### DISCUSSION TITLE
Hydrotherm X8 water heater configuration

### DISCUSSION THREAD

**Discussion by knowc** [make-all/tuya-local#1613] [Q&A]
I've recently installed a Hydrotherm X8 water heater, which I have seen other users report working well with Tuya Local.

I have attempted to follow the Configuration steps precisely. However at 'Stage Two' when selecting the device, the list does not include the Hydrotherm X8. (Please see below.)
![image](https://github.com/make-all/tuya-local/assets/39365859/fe244faa-01e6-486d-98dd-831e0a665773)
To state the obvious, I am not able to control the heater properly when configured as these other devices, although some sensors are reported correctly.

Previous users have reported no issues with the Gen 5 model; the manual/documentation that came with my heater indicates it's a Gen 6 model and I wonder if that might be my issue. (Although it could well be my mistake elsewhere.) Any help is much appreciated!

**Comment by make-all** [comment]
I don't know, you would have to provide the log message with dps listed that is output when that dialog in the screenshot is shown.

**Comment by knowc** [comment]
Thanks for looking into this, Jason.

I couldn't find any relevant log message when the dialog in my screenshot is shown. (I snooped around, but don't know where specifically to look.)

However, after watching a YouTube tutorial I did tabulate below my device's DP ID information as reported through iot.tuya.com. For non-numeric values, I have reported all the example values logged over five days but they may not be exhaustive. (e.g. I don't know if Four_valve_state can take values ON/OFF or something different like Water Flow which can be ON/Low Fan/High Fan. There are also more water heater Modes than the ones listed below.)

I hope this is useful information, otherwise I would appreciate any advice on how to provide the needed log message.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/knowc/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/knowc/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">


DPID | Description | Device Event | Example Values | unit
-- | -- | -- | -- | --
1 | Power | Report/Publish | ON; OFF | -
2 | Mode | Report/Publish | HYB; ECO; Standard | -
4 | Target Temp. | Report | 65 | ℃
7 | Defrosting | Report | OFF | -
14 | EEV Open | Report | 344 | P
15 | Fault Warning | Report | 0 |  
17 | Phase sequence module | Report | Nothing | -
18 | Today Electricity | Report | 0 | kW·h
19 | Water Flow | Report | OFF; Low Fan; High Fan | -
20 | Suction Temp. | Report | 7 | ℃
21 | Inlet Temp. | Report | 43 | ℃
22 | Outlet Temp. | Report | -50 | ℃
23 | Coiler Temp. | Report | 5 | ℃
24 | Exhaust Temp. | Report | 51 | ℃
26 | Ambient Temp. | Report | 18 | ℃
27 | Compressor State | Report | ON; OFF | -
28 | Four_valve_state | Report | OFF | -
29 | Low pressure switch | Report | Open | -
30 | Fluorine/Water | Report | Fluorine | -
31 | Pump State | Report | OFF | -
32 | Heating Element State | Report | ON; OFF | -
33 | High pressure switch | Report | Close | -
40 | Unit Tooling Number | Report | 5 | -
41 | Running time before defrost | Report | 546 | min
101 | Current | Report | 0.00 | A
102 | Voltage | Report | 0.00 | V
103 | Power | Report | 0.00 | W
104 | Total Electricity | Report | 0.00 | kW·h
105 | Electronic | Report | off | -
106 | Linked switch | Report | Open | -
107 | Water flow switch | Report | Open | -
108 | SW1 | Report | Open | -
109 | SW2 | Report | Open | -



</body>

</html>

**Reply by make-all** [reply]
The log message will be in the Home Assistant Core logs (Settings/System/Logs), provided you have not customised the logging to hide warning level messages.  After you find the log message, please file a new device request with all the requested details from the template, not just with example values, but with valid ranges for all integer and enum inputs.

**Reply by knowc** [reply]
I'm having an embarrassing time finding the needed information. The Home Assistant Core logs only contain the following relevant entry, without listing a dps:
`2024-02-06 20:50:12.953 WARNING (MainThread) [homeassistant.util.async_] Detected blocking call to sleep inside the event loop by custom integration 'tuya_local' at custom_components/tuya_local/device.py, line 77: self._api = tinytuya.Device(dev_id, address, local_key), please create a bug report at https://github.com/make-all/tuya-local/issues`

I'll keep poking around to see if I can get the needed details.

**Reply by make-all** [reply]
The log message will come when you get to the dialog that you sent a screenshot of at the top of the page.

The "sleep" message indicates that you have not filled in an IP address, and it has to wait for the device to show itself.
